### PR TITLE
Link styles in map

### PR DESCRIPTION
**Pull request by Wintxer**

### CODE REVIEW

**Comment by Wintxer** [comment]
TODO:
- Need an update link function
- Error 500 on getAllLinks when testing on a large study

File: webapp/src/components/App/Singlestudy/explore/Modelization/Map/index.tsx [hunk 1]
```diff
@@ -300,19 +301,18 @@ function Map() {
...
-                    )
-                  ),
-                [] as Array<LinkProperties>
-              )
+              links.map((link) => {
+                return {
+                  source: link.area1,
+                  target: link.area2,
+                  color: link.ui
+                    ? `rgb(${link.ui?.color.split(",").join(", ")}`
```
Comment (skamril) [inline webapp/src/components/App/Singlestudy/explore/Modelization/Map/index.tsx]: `split` and `join` not necessary

File: webapp/src/services/api/studydata.ts [hunk 2]
```diff
@@ -61,9 +61,12 @@ export const deleteLink = async (
 };
 
 export const getAllLinks = async (
-  uuid: string
+  uuid: string,
+  withUi?: boolean
 ): Promise<Array<LinkCreationInfo>> => {
```
Comment (skamril) [inline webapp/src/services/api/studydata.ts]: Use conditional types return (cf. `getUsers` in "src/services/api/users.ts").
You need to have `LinkCreationInfoDTO` and `LinkCreationInfoWithUiDTO` that extends the previous one.

**Review by skamril — CHANGES_REQUESTED** [review]

File: webapp/src/common/types.ts [hunk 3]
```diff
@@ -563,9 +563,16 @@ export interface UpdateAreaUi {
   color_rgb: Array<number>;
 }
 
+export interface LinkCreationUI {
```
Comment (skamril) [inline webapp/src/common/types.ts]: `DTO` suffix when it comes from server.
(Also for `LinkCreationInfo`)

**Review by skamril — CHANGES_REQUESTED** [review]

File: webapp/src/components/App/Singlestudy/explore/Modelization/Map/index.tsx [hunk 4]
```diff
@@ -300,19 +301,32 @@ function Map() {
...
+                  linecap = "round";
+                }
+                if (link.ui?.style === "dash") {
+                  style = [16, 8];
+                  linecap = "square";
+                }
+                if (link.ui?.style === "dotdash") {
+                  style = [10, 6, 1, 6];
+                  linecap = "square";
+                }
```
Comment (pl-buiquang) [inline webapp/src/components/App/Singlestudy/explore/Modelization/Map/index.tsx]: here you can use R.cond from Ramda (check examples in code) : https://ramdajs.com/docs/#cond

File: webapp/src/common/types.ts [hunk 5]
```diff
@@ -563,11 +563,21 @@ export interface UpdateAreaUi {
   color_rgb: Array<number>;
 }
 
-export interface LinkCreationInfo {
+export interface LinkCreationUIDTO {
```
Comment (pl-buiquang) [inline webapp/src/common/types.ts]: rename LinkUIInfoDTO

**Review by pl-buiquang — APPROVED** [review]

File: webapp/src/common/types.ts (hunk 5 above)
Comment (skamril) [reply webapp/src/common/types.ts]: `LinkInfoDTO` instead we have `LinkInfoDTOWithUI`

**Pull request merged**